style.lua

```diff
@@ -168,14 +210,35 @@ function filter_tags_relation_member (keyvalues, keyvaluemembers, roles, memberc
       -- Treat as polygon
       polygon = 1
       polytagcount = 0;
-      -- Count the number of polygon tags of the object
+      -- Count the number of polygon tags
+      -- First count keys in polygon_keys
       for i,k in ipairs(polygon_keys) do
          if keyvalues[k] then
+            polygontag = 1
+            -- However, if the key/value combination occurs in linestring_values, do not count the object as polygon
+            for index,tag in pairs(linestring_values) do
+               if k == tag[1] and keyvalues[k] == tag[2] then
+                  polygontag = 0
+                  break
+               end
+            end
+            if polygontag == 1 then
+               polytagcount = polytagcount + 1
+            end
+         end
+      end
+
+      -- Treat objects with a key/value combination in polygon_values as polygon
+      for index,tag in pairs(polygon_values) do
+         if keyvalues[tag[1]] == tag[2] then
             polytagcount = polytagcount + 1
+            break
```

Comment:
If you actually mean to count the polygon tags as it was done before, that break must be removed. However, I'd argue that the code was wrong before already. 'polytagcount' is only used to check if there are any polygon tags, so a simple boolean flag will do, just as above in the way function. So instead of removing this break, rather add one in line 226.